lib/models/store_transaction.dart

```diff
@@ -7,17 +7,41 @@ part 'store_transaction.g.dart';
 
 /// Represents a purchase transaction
 class StoreTransaction with _$StoreTransaction {
-  const factory StoreTransaction(
+  const factory StoreTransaction._create(
```

Comment:
Here we are adding a name `_create` to the "full" constructor, making it private, so it won't be accessible by devs. Then, we create a new unnamed factory pointing to this one with only the previous parameters. This way we won't break existing users of the previous constructor. 

We should note that new constructors won't be available for devs so:
- This system would break once we don't have just a basic rename of a property, as in, if we add a new property in the future, unless we want to make it nullable, the deprecated constructor won't know what value to use.
- People can still create instances through the `fromJson` constructor.
- We should consider making all constructors private in the next major.

Let me know what you think of this approach @RevenueCat/coresdk

Reply:
It makes sense to me, but I know virtually 0 about Dart so I'll leave others to review this.

Reply:
We got feedback in the past that having no constructors and only the `fromJson` constructor made testing hard since instantiating objects was cumbersome, so I am not sure this is the right approach to be honest, but I also don't have a better suggestion 🤔

Reply:
Hmm that is tricky since it would mean we can't ever modify these models easily. So either we break the constructor API or we add constructors for every new field with some default value, which might not be straightforward every time, or this solution, which could also break the fromJson constructor if we change the body and we don't handle missing parameters...

I'm thinking we could also make the constructor an experimental API to mark is as it could potentially change at anytime: https://api.flutter.dev/flutter/meta/experimental-constant.html . It's not great since it would still be breaking... But people would be more aware of it at least... Thoughts @vegaro ?

Reply:
Hmm so following the same process we did in Android (just add a new deprecated constructor) won't work in flutter. In dart you can only have one default/unnamed constructor. So I think if we want to have a public constructor with always up-to-date information, we can make the full constructor public and mark it as experimental as I suggested above. Lmk what you think @RevenueCat/coresdk

Reply:
yeah... I think that's probably the best to be honest

I was trying to think if maybe we could make instead of constructors, static functions that create objects. Like factories. So we could have a private constructor but have public factory methods (like the fromJson one) that accept the properties of the object as parameters. It would be easier to add new parameters to these factory methods.

Reply:
Well, that's kinda what factories do right? The thing is that to keep backwards compatibility, we need to keep the unnamed constructor be the one that existed until now (we can only have one). The latest changes are basically:
- Adding the unnamed factory constructor deprecated with the signature that existed until now
- Make the full constructor/factory name `create` with a comment indicating that it's subject to changes. So it would be used like `StoreTransaction.create(...)`.  

If we want to add more constructors in the future, we would need to add more factory methods with different names. Lmk what you think of this approach!